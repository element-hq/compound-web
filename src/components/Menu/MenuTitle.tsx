/*
 * Copyright 2025 New Vector Ltd
 *
 * SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
 * Please see LICENSE files in the repository root for full details.
 */

import React from "react";
import { Text } from "../Typography/Text.tsx";
import styles from "./MenuTitle.module.css";
import classnames from "classnames";

interface MenuTitleProps {
  /**
   * The title of the menu.
   */
  title: string;
  /**
   * The id of the menu title.
   */
  id?: string;
  /**
   * The CSS class.
   */
  className?: string;
}

export const MenuTitle: React.FC<MenuTitleProps> = ({
  title,
  id,
  className,
}) => {
  const classes = classnames(styles["menu-title"], className);

  return (
    <Text as="h3" id={id} className={classes} size="sm" weight="semibold">
      {title}
    </Text>
  );
};
