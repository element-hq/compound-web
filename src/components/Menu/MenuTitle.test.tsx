/*
 * Copyright 2025 New Vector Ltd
 *
 * SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
 * Please see LICENSE files in the repository root for full details.
 */

import { describe, expect, it } from "vitest";
import { render } from "@testing-library/react";
import { MenuTitle } from "./MenuTitle.tsx";
import React from "react";

describe("MenuTitle", () => {
  it("renders", () => {
    const { asFragment } = render(<MenuTitle title="Title" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
