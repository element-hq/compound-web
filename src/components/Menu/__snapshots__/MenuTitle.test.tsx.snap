// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MenuTitle > renders 1`] = `
<DocumentFragment>
  <h3
    class="_typography_489030 _font-body-sm-semibold_489030 _menu-title_7bbc38"
  >
    Title
  </h3>
</DocumentFragment>
`;
